### PR TITLE
Python 3 (with Py 2 compatibility) 

### LINKED ISSUE #2: Port to Python 3
https://discourse.numenta.org/t/proposal-to-introduce-pybind-for-move-toward-python-3-compatibility/3110/37 

Also Issue in #numenta/nupic 

Subtasks: 
- [ ] pybind migration https://github.com/chhenning/nupic/tree/master/src/cpp/nupic/python

### PR DESCRIPTION
**Pull request by pepedocs**
I've worked on top of @ctrl-z-9000-times work, here's what I've done so far.

Fixes:
 - Used `future` to allow py2 compatibility
 - Fixed install errors

With these changes nupic.py can now be installed with latest nupic bindings for both py2 and py3. There are still some test (runtime) errors that are related to py2 compat and also incorrect usage of `nupic.cpp` api.

Please review @ctrl-z-9000-times @breznak 

Fixes #2

### CODE REVIEW

**Comment by breznak** [comment]
I'm attaching a nice cheatsheet which basically guides through the py2, 3 compatible rewrite. 
https://python-future.org/compatible_idioms.html

**Review by breznak — CHANGES_REQUESTED** [review]
Looks good to me! :clap: Thank you, guys! :+1: 

> are still some test (runtime) errors that are related to py2 compat 

please fix these

> and also incorrect usage of nupic.cpp api.

while we'll ignore any of these

Meanwhile, I'll have a look at fixing the CI for this repo.

File: requirements.txt [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+coverage>=3.7.1
+mock>=1.0.1
+ordereddict>=1.1
+psutil>=1.0.1
+pytest>=3.0.7
+pytest-cov>=2.5.0
+pytest-xdist>=1.16.0
+python-dateutil>=2.1
+PyYAML>=3.10
+#unittest2>=0.5.1
```
Comment (breznak) [inline requirements.txt:12]: this PR uses `unittest` while PR #5 suggests existing `unittest2` updated to >= 1.10. 
What is the difference?

**Review by breznak — COMMENTED** [review]
Python 3 will still fail is numenta's bindings do not support it.

**Comment by breznak** [comment]
Only one failing test: 
https://travis-ci.com/htm-community/nupic.py/jobs/180565542#L710

**Review by D-McDougall — APPROVED** [review]
This looks good to me.  Thank you!

**Review by breznak — APPROVED** [review]

**Comment by breznak** [comment]
I am merging as is, so we can continue the development.

**Comment by pepedocs** [comment]
> Only one failing test:
> https://travis-ci.com/htm-community/nupic.py/jobs/180565542#L710

I believe this error occurs only for py3? I've got a fix for this, also there are more py2 errors I just have to fix them first.

**Pull request merged**